### PR TITLE
✨ feat: Pass user input to uart

### PR DESCRIPTION
**Pull request by Teufelchen1**
Adds a text input to the tui. The input is passed to the mmap-uart.
Additionally the C test program makes use of that feature.

### CODE REVIEW

File: src/ui.rs [hunk 1]
```diff
@@ -183,3 +220,85 @@ fn centered_rect(percent_x: u16, percent_y: u16, r: Rect) -> Rect {
...
+    let mut insert_mode = false;
+    let mut auto_step = false;
+
+    loop {
+        terminal.draw(|f| ui.ui(f, cpu, uart_rx, show_help, insert_mode, &user_input))?;
+
+        if let Ok(maybe_event) = event::poll(Duration::from_millis(50)) {
+            if maybe_event {
+                if let Event::Key(key) = event::read()? {
+                    if insert_mode {
```
Comment (Teufelchen1) [inline src/ui.rs]: @edjopato I have a lot of nesting here. Do you think there is a more elegant way of doing the `poll & query & matching`?

Comment (Teufelchen1) [reply src/ui.rs]: (and please don't invest too much time here, this is likely to get refactored anyways soonish)

Comment (EdJoPaTo) [reply src/ui.rs]: Maybe with [`Result::is_ok_and`](https://doc.rust-lang.org/std/result/enum.Result.html#method.is_ok_and)? Or just `poll(…)?` / `poll(…).expect(…)`? Event handling is a core feature required, when it doesn't work the application is kinda pointless? Then its just `if event::poll(…)? {…}`.

The `if insert_mode` seems needed for the logic.

When event handling starts to involve something else than keys the `event::read()` can be better handled by a match directly. (`if event::poll(…)? { match event::read()? { Event::Key(KeyCode::Enter)) => {…} } }`. Not sure whether thats a benefit in your case…

Comment (Teufelchen1) [reply src/ui.rs]: Thanks 💛!
Your hint towards `if event::poll(…)? {…}` did the trick!

**Pull request merged**